packages/@aws-cdk/core/lib/secret-value.ts

```diff
@@ -96,14 +102,25 @@ export class SecretValue extends Intrinsic {
 
     return new SecretValue(param.value);
   }
+
+  /**
+   * Specifies the secret qualifiers when they were provided for a value from SecretsManager
+   */
+  readonly secretQualifier?: SecretQualifier;
+
+  constructor(value: any, props: SecretValueProps = {}) {
```

Comment:
Looks like our backwards-compatibility checker is having some problems with this change.

Can you try this, and see if that helps?

```suggestion
  constructor(value: any, props?: IntrinsicProps)
  constructor(value: any, props: SecretValueProps = {}) {
```

Reply:
This does not seem to work inside with the changes we made in `SecretValue.secretsManager`.

Reply:
This does not seem to work inside with the changes we made in `SecretValue.secretsManager`.

Reply:
What's the error?

Reply:
It complains about this:
https://github.com/aws/aws-cdk/blob/cdf02649bb589e447f84ed52b6f68ed5932b5edd/packages/%40aws-cdk/core/lib/secret-value.ts#L65
```
lib/secret-value.ts:65:37 - error TS2345: Argument of type '{ secretQualifier: SecretQualifier; }' is not assignable to parameter of type 'IntrinsicProps'.
  Object literal may only specify known properties, and 'secretQualifier' does not exist in type 'IntrinsicProps'.
```

Reply:
OK, let's try one last trick before we give up. Let's try to split the props into a separate variable:

```ts
    const secretQualifier: SecretQualifier = {
      secretId,
      jsonField: options.jsonField,
      versionStage: options.versionStage,
      versionId: options.versionId,
    };
    const props = { secretQualifier }; // might require a type annotation (props: SecretValueProps)
    return new SecretValue(dyref, props);
```

Reply:
This worked like a charm!